source/gui/settingsDialogs.py

```diff
@@ -3195,6 +3195,8 @@ class DictionaryDialog(SettingsDialog):
 	helpId = "SpeechDictionaries"
 
 	def __init__(self,parent,title,speechDict):

```

Comment:
Have you considered making the constructor abstract rather than checking type of the dialog? This would be way more Pythonic IMO.